Failed: NewConnectionPoolSpec should open a second connection if the first one is loaded

https://github.com/akka/akka-http/runs/5410042144?check_suite_focus=true

```
2022-03-03T16:18:28.4997872Z [info] - should open a second connection if the first one is loaded *** FAILED *** (10 seconds, 51 milliseconds)
2022-03-03T16:18:28.4998473Z [info]   java.lang.AssertionError: Expected OnNext(_), yet no element signaled during 10 seconds
2022-03-03T16:18:28.4999076Z [info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:347)
2022-03-03T16:18:28.4999692Z [info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:338)
2022-03-03T16:18:28.5000407Z [info]   at akka.http.impl.engine.client.NewConnectionPoolSpec$TestSetup.acceptIncomingConnection(ConnectionPoolSpec.scala:678)
2022-03-03T16:18:28.5001537Z [info]   at akka.http.impl.engine.client.NewConnectionPoolSpec$$anon$2.<init>(ConnectionPoolSpec.scala:100)
2022-03-03T16:18:28.5002226Z [info]   at akka.http.impl.engine.client.NewConnectionPoolSpec.$anonfun$new$3(ConnectionPoolSpec.scala:91)

```

--- a/akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala
+++ b/akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala
@@ -89,15 +89,29 @@ class NewConnectionPoolSpec extends AkkaSpecWithMaterializer("""
     }
 
     "open a second connection if the first one is loaded" in new TestSetup {
+
+      case class Request(request: HttpRequest, connNo: Int, promise: Promise[HttpResponse]) {
+        def handle(): Unit = promise.success(testServerHandler(connNo)(request))
+      }
+      override def asyncTestServerHandler(connNr: Int): HttpRequest => Future[HttpResponse] = { req =>
+        val p = Promise[HttpResponse]()
+        testActor ! Request(req, connNr, p)
+        p.future
+      }
+
       val (requestIn, responseOut, responseOutSub, _) = cachedHostConnectionPool[Int]()
+
+      responseOutSub.request(2)
 
       requestIn.sendNext(HttpRequest(uri = "/a") -> 42)
       requestIn.sendNext(HttpRequest(uri = "/b") -> 43)
 
-      responseOutSub.request(2)
       acceptIncomingConnection()
+      acceptIncomingConnection()
+
+      expectMsgType[Request].handle()
       val r1 = responseOut.expectNext()
-      acceptIncomingConnection()
+      expectMsgType[Request].handle()
       val r2 = responseOut.expectNext()
 
       Seq(r1, r2) foreach {

